cluster/gce/manifests/pdcsi-controller.yaml

```diff
@@ -0,0 +1,165 @@
+    - mountPath: /etc/srv/kubernetes/pdcsi-controller
+      name: srv
+      readOnly: true
+  - args:
+    - --v=5
+    - --csi-address=/csi/csi.sock
+    - --http-endpoint=:22012
+    - --leader-election
+    - --leader-election-namespace=$(PDCSI_NAMESPACE)
+    - --timeout=250s
+    - --kubeconfig=/etc/srv/kubernetes/pdcsi-controller/kubeconfig
+    env:
+    - name: PDCSI_NAMESPACE
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.namespace
+    image: k8s.gcr.io/sig-storage/csi-attacher:v3.2.1
+    livenessProbe:
+      failureThreshold: 1
+      httpGet:
+        path: /healthz/leader-election
+        port: http-endpoint
+      initialDelaySeconds: 10
+      periodSeconds: 20
+      timeoutSeconds: 10
+    name: csi-attacher
+    ports:
+    - containerPort: 22012
+      name: http-endpoint
+      protocol: TCP
+    volumeMounts:
+    - mountPath: /csi
+      name: socket-dir
+    - mountPath: /etc/srv/kubernetes/pdcsi-controller
+      name: srv
+      readOnly: true
+  - args:
+    - --v=5
+    - --csi-address=/csi/csi.sock
+    - --http-endpoint=:22013
+    - --leader-election
+    - --leader-election-namespace=$(PDCSI_NAMESPACE)
+    - --handle-volume-inuse-error=false
+    - --kubeconfig=/etc/srv/kubernetes/pdcsi-controller/kubeconfig
+    env:
+    - name: PDCSI_NAMESPACE
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.namespace
+    image: k8s.gcr.io/sig-storage/csi-resizer:v1.2.0
+    livenessProbe:
+      failureThreshold: 1
+      httpGet:
+        path: /healthz/leader-election
+        port: http-endpoint
+      initialDelaySeconds: 10
+      periodSeconds: 20
+      timeoutSeconds: 10
+    name: csi-resizer
+    ports:
+    - containerPort: 22013
+      name: http-endpoint
+      protocol: TCP
+    volumeMounts:
+    - mountPath: /csi
+      name: socket-dir
+    - mountPath: /etc/srv/kubernetes/pdcsi-controller
+      name: srv
+      readOnly: true
+  - args:
+    - --v=5
+    - --csi-address=/csi/csi.sock
+    - --metrics-address=:22014
+    - --leader-election
+    - --leader-election-namespace=$(PDCSI_NAMESPACE)
+    - --timeout=300s
+    - --kubeconfig=/etc/srv/kubernetes/pdcsi-controller/kubeconfig
+    env:
+    - name: PDCSI_NAMESPACE
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.namespace
+    image: k8s.gcr.io/sig-storage/csi-snapshotter:v3.0.3
+    name: csi-snapshotter
+    volumeMounts:
+    - mountPath: /csi
+      name: socket-dir
+    - mountPath: /etc/srv/kubernetes/pdcsi-controller
+      name: srv
+      readOnly: true 
+  - args:
+    - --v=5
+    - --endpoint=unix:/csi/csi.sock
+    - --cloud-config=/etc/gce.conf
+    - --run-node-service=false
+    image: k8s.gcr.io/cloud-provider-gcp/gcp-compute-persistent-disk-csi-driver:v1.3.1
+    name: gce-pd-driver
+    volumeMounts:
+    - mountPath: /csi
+      name: socket-dir
+    - mountPath: /etc/gce.conf
+      name: cloud-config-mount
+      readOnly: true
+  hostNetwork: true
```

Comment:
Does this need to be true. Should think how to document the range of ports we are reserving for it.

Reply:
(offline conversation wtih @diversario 

Since we aren't scraping metrics in this GCE configuration, it should be fine to switch off hostNetwork. @leiyiz if you remove this & the e2e tests pass we should be good to go :-)

Reply:
turning off the `hostNetwork` flag led to issue where containers are having trouble talking to the cluster. the pdcsi-controller is using a kubeconfig generated [here](https://github.com/kubernetes/cloud-provider-gcp/blob/master/cluster/gce/gci/configure-helper.sh#L1276) and that points to localhost, which we think only works if we have hostNetwork on.